link_shorterner/src/shortener_link_handler.erl

```diff
@@ -0,0 +1,37 @@
+-module(shortener_link_handler).
+
+-export([init/2]).
+
+init(Req, State) ->
+    Url = get_request_url(Req),
+    ReqMethod = cowboy_req:method(Req),
+    {Status, ShortUrl} = handle_request(ReqMethod, Url),
+    Body = get_body_response(Status, ShortUrl),
```

Comment:
I would suggest rearranging these three lines (get body, get headers and reply), in order to separate body building from encoding format. You most likely want to set the json content type in the same helper that does the json encoding.

Maybe something in the lines of

```
Body = #{<<"url">> => ShortUrl},
Resp = json_reply(Status, Body, Req)
```

Reply:
BTW, when building maps for later json encoding you may consider using atoms for keys instead of binaries (they are treated the same by the ecoding library and generally look better).